### PR TITLE
Quest translation deux

### LINKED ISSUE #9080: [BUG] Missing translation interface for mod's quest text
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.19.2

### MineColonies Version

1.0.1371-ALPHA

### Structurize Version

1.0.491-ALPHA

### Related Mods and their Versions

_No response_

### Current Behavior

1.0.1371-ALPHA version adds some quests, but I can't translate them on Crowdin, because Crowdin doesn't have them.
I can only override them if I want to translate. There's no second way. Once the mod updated the newer version, I have to override them again, that's too inconvenience.

### Expected Behavior

I hope quest texts can be translated to other language on Crowdin. 

### Reproduction Steps

1. Open `minecolonies.jar` by 7z;
2. Open `data\minecolonies\quests` folders, I found some json files;
3. These texts are all English.

### Logs

Not crash

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by uecasm**
Closes #9080

# Changes proposed in this pull request:
- Fixes #9085 (vanilla regex bug breaks `%d` in translation resources)
- Hotfix nullptr
- Fix resource file exclusions:
    - The build script was trying to exclude `default.json` from the jar, but didn't.  Now it does.
    - It also now excludes the new `quests.json` and the old `manual_en_us.json`.
    - This leaves `entries.json` and `en_us.json`, identical files.  I'm not sure why the first one exists.  Something crowdin related maybe?  But does it need to go in the jar?
- Magic translation datagen for quest json
    - The quest jsons themselves are still exactly as before.
    - During datagen it processes these files and generates translation keys for the dialogue automatically.  (This does unfortunately mean they have more generic names than if done manually, but this does still allow following the conversation path in the translation file.)
    - It doesn't make any attempt to de-duplicate the strings.

Review please (do not port, datagen will need some code changes for 1.19.3+)

### CODE REVIEW

**Comment by marchermans** [comment]
I am not sure about this PRs approach.
Does it not make sense to implement datagen once and for all properly? And handle it that way? Instead
of this backwards way where you make a "magic" translation datagen only?

**Comment by Raycoms** [comment]
I think quests are too unique and code writing dialogues is kinda a big nono

Sent from Proton Mail mobile

-------- Original Message --------

**Comment by uecasm** [comment]
My main thought here is that we seem to be wanting the community to write a lot of quests, and that seems like it would be easier with self-contained json files rather than with codegen.

Although a counterargument might be that the codegen could be written in a sufficiently fluent style that a non-coder would be able to follow it.  And that might let them avoid json's quirks.  Though modpack authors would have to use the json format regardless.

**Comment by marchermans** [comment]
I understand. It is a balancing act I agree.
Maybe ask some people in the community what they want.

**Comment by Raycoms** [comment]
> I understand. It is a balancing act I agree. Maybe ask some people in the community what they want.

They would optimally want to send us text files, but that would then put a lot of load on us, json files are okay too. The format is pretty easy to grasp for everything put the triggers

**Comment by marchermans** [comment]
> > I understand. It is a balancing act I agree. Maybe ask some people in the community what they want.
> 
> They would optimally want to send us text files, but that would then put a lot of load on us, json files are okay too. The format is pretty easy to grasp for everything put the triggers

Oke, then I am not sure what to do.....
A webservice?

**Comment by Raycoms** [comment]
> > > I understand. It is a balancing act I agree. Maybe ask some people in the community what they want.
> > 
> > 
> > They would optimally want to send us text files, but that would then put a lot of load on us, json files are okay too. The format is pretty easy to grasp for everything put the triggers
> 
> Oke, then I am not sure what to do..... A webservice?

Nah, they will give us the jsons and we include the jsons and datagen the translations from there then.

**Comment by uecasm** [comment]
We could come up with some third vaguely-plain-text file format, parse _that_, and datagen it into json and translations.

But I don't think that really gains us much over just the jsons, especially as (like the pure codegen) it's still _different_ from what modpacks would have to do anyway.  And adventure/gating modpacks are likely to be the most interested in creating quests, aside from whatever's in the mod itself.

File: src/main/java/com/minecolonies/coremod/generation/defaults/QuestTranslationProvider.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+
+import static com.minecolonies.api.quests.QuestParseConstant.*;
+import static com.minecolonies.api.quests.registries.QuestRegistries.DIALOGUE_OBJECTIVE_ID;
+import static com.minecolonies.api.util.constant.Constants.MOD_ID;
+import static com.minecolonies.coremod.quests.QuestParsingConstants.*;
+
+public class QuestTranslationProvider implements DataProvider
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/generation/defaults/QuestTranslationProvider.java:31]: needs javadoc

**Review by Raycoms — COMMENTED** [review]
Just needs a bit of javadoc

**Pull request merged**